src/worker.cpp

```diff
@@ -1242,7 +1242,11 @@ void Worker::collectFrusFromJson()
         const std::string& vpdFilePath = itemFRUS.key();
 
         // skip processing of system VPD again as it has been already collected.
-        if (vpdFilePath == SYSTEM_VPD_FILE_PATH)
+        // Also, if chassis is powered on, skip collecting FRUs which are
+        // powerOffOnly.
```

Comment:
Add a todo,
Need to revisit for P-Future to reduce code update time.

Reply:
Done.